common/lib/client/auth.js

```diff
@@ -260,7 +259,13 @@ var Auth = (function() {
 						}
 					}
 					cb(null, body);
-				});
+				};
+				Logger.logAction(Logger.LOG_MICRO, 'Auth.requestToken().tokenRequestCallback', 'Sending; ' + authOptions.authUrl + '; Params: ' + JSON.stringify(authParams));
+				if(authOptions.authMethod && authOptions.authMethod.toLowerCase() === 'post') {
+					Http.postUri(rest, authOptions.authUrl, authHeaders || {}, authParams, {}, authUrlRequestCallback);
```

Comment:
authParams must be capable of being included as query params so I think it is completely reasonable that we only support string values.

And yes, I think POST should default to form encoding until we add (in a later spec version) something that allows the client to specify the content type.